### PR TITLE
AVD test locally

### PR DESCRIPTION
**Pull request by ajinabraham**
This PR add support for using AVD with MobSF by @matandobr 
Code QA and Refactoring of Android Dynamic Analysis
Added functionality to remove scan results

### CODE REVIEW

File: DynamicAnalyzer/views/android/android_dynamic.py [hunk 1]
```diff
@@ -387,24 +388,42 @@ def mobsf_ca(request):
...
+                    # For some reason, avd emulator does not have cp binary
+                    subprocess.call([adb,
+                                     "-s",
+                                     get_identifier(),
+                                     "shell",
+                                     "/data/local/tmp/busybox",
+                                     "cp",
+                                     "/data/local/tmp/" + settings.ROOT_CA,
+                                     "/system/etc/security/cacerts/" + settings.ROOT_CA])
+                    subprocess.call([adb,
```
Comment (ajinabraham) [inline DynamicAnalyzer/views/android/android_dynamic.py]: Is `su` not required?

File: mobsfy_AVD.py [hunk 2]
```diff
@@ -214,7 +219,7 @@ def main():
         print_log("Emulator INI already exists")
     else:
         shutil.copyfile(emulator_ini, new_emulator_ini)
-    if os.path.isdir(avd_path):
+    if os.path.isdir(new_emulator_avd):
```
Comment (ajinabraham) [inline mobsfy_AVD.py]: We need to overwrite any files that we copy to `sdk` and `avd` directory. Also please give a graceful message that we will overwrite existing files. The overwrite is necessary as we may release avd image updates in future.

File: templates/dynamic_analysis/start_test.html [hunk 3]
```diff
@@ -194,6 +194,10 @@ <h2 class="page-header"><strong>Dynamic Analyzer </strong>- {{ pkg }}</h2>
                                     	$('#start').text("Environment Created");
                                     	$('#stat').append("\nRunning HTTPS Proxy");
 
+                                    {% comment %}
+                                        Avd has mobsfCA prebuilt
+                                    {% endcomment %}
+                                    {% if AVD != True %}
```
Comment (ajinabraham) [inline templates/dynamic_analysis/start_test.html]: Should add.
```
 {% if AVD == True %}
 $('#stat').append("\nMobSF RootCA Installed Successfully");
 {% endif %}
```

File: mobsfy_AVD.py (hunk 2 above)
Comment (ajinabraham) [inline mobsfy_AVD.py]: We need to overwrite any files that we copy to sdk and avd directory. Also please give a graceful message that we will overwrite existing files. The overwrite is necessary as we may release avd image updates in future.

**Comment by ajinabraham** [comment]
Checklist Before Committing to Master
- [x] Code Review
- [ ] Bug Fixes
- [ ] Testing in OSX
- [ ] Testing in Ubuntu
- [ ] Testing in Kali
- [ ] Testing in Windows

File: templates/dynamic_analysis/start_test.html (hunk 3 above)
Comment (matandobr) [inline templates/dynamic_analysis/start_test.html]: done

File: DynamicAnalyzer/views/android/android_dynamic.py (hunk 1 above)
Comment (matandobr) [inline DynamicAnalyzer/views/android/android_dynamic.py]: Nope, adb shell commands runs as root

Comment (ajinabraham) [inline DynamicAnalyzer/views/android/android_dynamic.py]: Ok

**Comment by ajinabraham** [comment]
The latest comment refactored android dynamic analysis, made the process consistent. We need to update wiki. also fixed fixed the template to show MobSF RootCA installed for AVD.

**Pull request merged**